desc/coils.py

```diff
@@ -837,11 +1028,85 @@ def compute_magnetic_field(
         # coils curvature which is a 2nd derivative of the position, and doing that
         # with only possibly c1 cubic splines is inaccurate, so we don't do it
         # (for now, maybe in the future?)
-        B = biot_savart_hh(coords, coil_pts_start, coil_pts_end, current)
+        AB = op(coords, coil_pts_start, coil_pts_end, current)
 
         if basis == "rpz":
-            B = xyz2rpz_vec(B, x=coords[:, 0], y=coords[:, 1])
-        return B
+            AB = xyz2rpz_vec(AB, x=coords[:, 0], y=coords[:, 1])
+        return AB
+
+    def compute_magnetic_field(
```

Comment:
Isn't this already implemented by the parent `_Coil` class?

Reply:
same comment as Rory below, for splined coils the Hanson-Hirshman formula is preferred so we override in the subclass here